### ISSUE TITLE
Update emscripten [2 XMR]

### ISSUE THREAD

**Issue by woodser** [woodser/monero-ts#144]
This issue requests updating the build to use the latest version of [emscripten](https://emscripten.org/).

The library is currently using version 3.1.10.

**Comment by woodser** [comment]
See the work in progress draft PR as a starting point: https://github.com/woodser/monero-ts/pull/192

**Comment by NorrinRadd** [comment]
I will take this. Please assign.

**Comment by woodser** [comment]
Great, I've assigned to you @NorrinRadd

**Comment by woodser** [comment]
I forgot to mention, the upgrade should not use multithreading or pthreads, like I was experimenting with in the [draft PR](https://github.com/woodser/monero-ts/pull/192). Instead, it should still be single-threaded.

**Comment by NorrinRadd** [comment]
I have all the builds working except for with the new version of emscripten.  

To future-proof it, threads seem important.  Threads were disabled somehow with the previous version? @woodser

**Comment by woodser** [comment]
But if you can get it working with threads (since the new emscripten seems to really prefer that), then that's great progress, and we could attempt to remove threads after.

**Comment by NorrinRadd** [comment]
yeah the monero project seems to require threads: https://github.com/monero-project/monero/blob/master/src/wallet/node_rpc_proxy.h#L32

**Comment by woodser** [comment]
This project builds and runs now without threads, so no, it isn't strictly necessary.

**Comment by NorrinRadd** [comment]
@woodser i got it to build with the latest emscripten. Attempting to run tests now. Your `npm test` requires that testnet be fully synced and not pruned?

**Comment by NorrinRadd** [comment]
@woodser PR submitted

**Comment by NorrinRadd** [comment]
Along with #206, 
* isolated it down to the build_boost_emscripten phase
* downgraded boost to a version that was out around when emscripten 3.1.26 (last working version) was out >> still fails. Therefore the culprit seems to be emscripten.
* sent a bug report to emscripten regarding the newer versions: https://github.com/emscripten-core/emscripten/issues/21793 Hopefully can get some transaction soon.

**Comment by NorrinRadd** [comment]
@woodser Update from emscripten team: They're of the belief that this is deliberate from Boost; that if certain classes are used, then multi-threading must be enabled. It also seems like it was an error that Emscripten < 3.1.26 was avoiding this somehow, but I have not received a reply from them on that yet. 

```
ignoring nonexistent directory "/Users/user/code/emsdk/upstream/emscripten/cache/sysroot/include/ wasm32-emscripten/c++/v1"
ignoring nonexistent directory "/Users/user/code/emsdk/upstream/emscripten/cache/sysroot/include/ wasm32-emscripten"
#include "..." search starts here:
#include <...> search starts here:
 .
 /Users/user/code/emsdk/upstream/emscripten/cache/sysroot/include/fakesdl
 /Users/user/code/emsdk/upstream/emscripten/cache/sysroot/include/compat
 /Users/user/code/emsdk/upstream/emscripten/cache/sysroot/include/c++/v1
 /Users/user/code/emsdk/upstream/lib/clang/19/include
 /Users/user/code/emsdk/upstream/emscripten/cache/sysroot/include
End of search list.
In file included from libs/thread/src/pthread/thread.cpp:9:
In file included from ./boost/thread/detail/config.hpp:13:
In file included from ./boost/thread/detail/platform.hpp:17:
./boost/config/requires_threads.hpp:29:4: error: "Threading support unavaliable: it has been      explicitly disabled with BOOST_DISABLE_THREADS"
   29 | #  error "Threading support unavaliable: it has been explicitly disabled with             BOOST_DISABLE_THREADS"
      |    ^   
1 error generated.
```
You can check the first PR (regarding multi-threading support) that it has been updated with a fix for the shared buffer issue, as well as a web worker issue.  The remainder of the work seems to be updating monero-ts to actually use the web workers.

**Comment by woodser** [comment]
Thanks for the update. Maybe multithreading is the way forward, though will require changes to any downstream dependents.

**Comment by NorrinRadd** [comment]
@woodser, Have you seen the update that denotes that Emscripten 3.1.26 can
be used? It gets the upgrade part of the way there.

**Comment by NorrinRadd** [comment]
@woodser please what is the update? emscripten has been updated as far as possible. Please evaluate this pull request.

**Comment by woodser** [comment]
Ideally we can update to the latest emscripten, built with pthreads, but with pthreads disabled as a final step in the linker flags, before a major update to support real threading.

It's limited use to update halfway.

**Comment by woodser** [comment]
Resolved by @mainnet-pat in https://github.com/woodser/monero-ts/pull/258

**Comment by NorrinRadd** [comment]
@woodser I'm just now noticing that [PR#206](https://github.com/woodser/monero-ts/pull/206) was partially merged in [PR#224](https://github.com/woodser/monero-ts/pull/224/). It was also used as research for [PR#258](https://github.com/woodser/monero-ts/pull/258). Given that, will some of the bounty be paid?
